crates/uv/src/cli.rs

```diff
@@ -1945,6 +1945,11 @@ pub(crate) struct ToolRunArgs {
     #[arg(allow_hyphen_values = true)]
     pub(crate) args: Vec<OsString>,
 
+    // TODO(zanieb): Consider using `Vec<String>` instead, treating the first as the target package
+    /// Use the given package specification instead of inferring it from the target command.
+    #[arg(long)]
+    pub(crate) package: Option<String>,
```

Comment:
Open to some bike-shedding here.

pipx says

> --spec SPEC           
>
> The package name or specific installation source passed to pip. Runs `pip install -U SPEC`. For
> example `--spec mypackage==2.0.0` or `--spec git+https://github.com/user/repo.git@branch`

which I find very unintuitive

Reply:
Hmm. `package` isn't bad but it might be confusing with `-p` in the workspace API. I find `spec` okay but don't feel strongly.

Reply:
Like `-p` being short for `--python`?

Reply:
Oh, I was assuming `-p` would be short for package in the workspace API, as it is in cargo (`cargo test -p uv`, etc.).

Reply:
Uh oh haha it's short for `--python` right now. We might need to reconsider that?

I'll think about this some more.

Reply:
`--from`? e.g. `uv tool run --from foo -- bar` or `uv tool run --from git+https://github.com/user/repo.git@branch -- bar`

Reply:
If you use `--with` (as in #3678), we will still infer the package that provides the tool from the command name so `--with` is additive while this flag _replaces_ the inferred requirement.

I considered having the first `--with` package replace the inferred requirement but that seems confusing.

Reply:
This is especially important for installation where we won't install entry points from additional dependencies, just the tool package.